pkg/services/ngalert/image/service.go

```diff
@@ -0,0 +1,112 @@
+package image
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/grafana/grafana/pkg/components/imguploader"
+	ngmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/store"
+	"github.com/grafana/grafana/pkg/services/rendering"
+	"github.com/grafana/grafana/pkg/services/screenshot"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+//go:generate mockgen -destination=mock.go -package=image github.com/grafana/grafana/pkg/services/ngalert/image ImageService
+type ImageService interface {
+	// NewImage returns a new image for the alert instance.
+	NewImage(ctx context.Context, r *ngmodels.AlertRule, labels data.Labels) (*store.Image, error)
+}
+
+var (
+	// ErrNoDashboard is returned when the alert rule does not have a dashboard.
+	ErrNoDashboard = errors.New("no dashboard")
+
+	// ErrNoPanel is returned when the alert rule does not have a panel in a dashboard.
+	ErrNoPanel = errors.New("no panel")
+)
+
+const (
+	screenshotTimeout  = 10 * time.Second
+	screenshotCacheTTL = 15 * time.Second
+)
+
+// ScreenshotImageService takes screenshots of the panel for an alert rule and
+// saves the image in the store. The image contains a unique token that can be
+// passed as an annotation or label to the Alertmanager.
+type ScreenshotImageService struct {
+	screenshots screenshot.ScreenshotService
+	store       store.ImageStore
+}
+
+func NewScreenshotImageService(screenshots screenshot.ScreenshotService, store store.ImageStore) ImageService {
+	return &ScreenshotImageService{
+		screenshots: screenshots,
+		store:       store,
+	}
+}
+
+// NewScreenshotImageServiceFromCfg returns a new ScreenshotImageService
+// from the configuration.
+func NewScreenshotImageServiceFromCfg(cfg *setting.Cfg, r prometheus.Registerer,
+	db *store.DBstore, rs rendering.Service) (ImageService, error) {
+	if !cfg.UnifiedAlerting.Screenshots.Enabled {
+		return &ScreenshotImageService{
```

Comment:
**TODO**: We don't want error messages just because screenshots are disabled

Reply:
I'm handling this in state/manager.go's newImage function.